sw/device/lib/crypto/impl/ecc/ecdh_p384.h

```diff
@@ -0,0 +1,78 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_CRYPTO_IMPL_ECC_ECDH_P384_H_
+#define OPENTITAN_SW_DEVICE_LIB_CRYPTO_IMPL_ECC_ECDH_P384_H_
+
+#include <stddef.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/hardened.h"
+#include "sw/device/lib/crypto/drivers/otbn.h"
+#include "sw/device/lib/crypto/impl/ecc/p384_common.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+/**
+ * A type that holds a blinded ECDH shared secret key.
+ *
+ * The key is boolean-masked (XOR of the two shares).
+ */
+typedef struct ecdh_p384_shared_key {
+  uint32_t share0[kP384CoordWords];
+  uint32_t share1[kP384CoordWords];
+} ecdh_p384_shared_key_t;
+
+/**
+ * Start an async ECDH/P-384 keypair generation operation on OTBN.
+ *
+ * Returns an `OTCRYPTO_ASYNC_INCOMPLETE` error if OTBN is busy.
+ *
+ * @return Result of the operation (OK or error).
+ */
+status_t ecdh_p384_keypair_start(void);
```

Comment:
Add this macro to ensure that callers get warnings if they don't check the error code:
```suggestion
OT_WARN_UNUSED_RESULT
status_t ecdh_p384_keypair_start(void);
```
Same goes for every other function declaration added in this PR that returns `status_t`. Even internal functions declared with `static` should get this macro to help protect against callers not forwarding failures.

Reply:
Sure, makes sense. But, as far as I see, this then would also concern a lot of other files which aren't even affected by this PR.

Reply:
Then those files have errors, but I still don't want to merge new errors in when code is being added. I've been trying to fix these missing macros every time I see them, even if I'm just vaguely editing code in the same file.